UI/FleetWnd.cpp

```diff
@@ -3532,29 +3528,18 @@ void FleetWnd::FleetRightClicked(GG::ListBox::iterator it, const GG::Pt& pt, con
     {
         auto split_action = [this, &ship_ids_set]() {
             ScopedTimer split_fleet_timer("FleetWnd::SplitFleet", true);
-            // remove first ship from set, so it stays in its existing fleet
-            auto ship_id_it = ship_ids_set.begin();
-            ship_ids_set.erase(ship_id_it);
-
-            FleetAggression new_aggression_setting = FleetAggression::INVALID_FLEET_AGGRESSION;
-            if (m_new_fleet_drop_target)
-                new_aggression_setting = m_new_fleet_drop_target->GetFleetAggression();
```

Comment:
I don't think there'll ever be a drop target when a fleet is split. Same in split_per_design below.

Reply:
This isn't checking if the drop target was used as part of the splitting action, but rather if there is a drop target in the window. It can be present without being part of the action that leads to the check.

Reply:
Ah! I've never used that feature, but I could see how it works now. Will revert that part.